src/Client.re

```diff
@@ -234,21 +241,23 @@ let executeQuery =
       (),
     ) => {
   let req =
-    createRequest(~query=request##query, ~variables=request##variables, ());
-  let parse = request##parse;
-  let optsJs =
-    Types.partialOperationContext(
-      ~additionalTypenames?,
-      ~fetchOptions?,
-      ~fetch?,
-      ~requestPolicy=?Belt.Option.map(requestPolicy, Types.requestPolicyToJs),
-      ~url?,
-      ~pollInterval?,
-      ~meta?,
-      ~suspense?,
-      ~preferGetMethod?,
+    Utils.createRequest(
+      ~query=request##query,
+      ~variables=request##variables,
       (),
     );
+  let parse = request##parse;
+  let optsJs: Types.partialOperationContext = {
+    additionalTypenames,
+    fetchOptions,
+    fetch,
+    requestPolicy: requestPolicy->Belt.Option.map(Types.requestPolicyToJs),
+    url: Some(url->Belt.Option.getWithDefault(client.url)),
```

Comment:
This makes me 😭 With our labeled argument being passed into this function, we allow users to overwrite the `url` that `executeQuery` requests. This is good! However, when composing this object, if a user _doesn't_ supply the `url` parameter, the `context` object will get compiled to have `url: undefined`. This then overwrites the `url` supplied by the Client, since `context` is spread into `executeQuery`. So we ensure here that we safely fallback to `client.url` if no `url` is provided. Annoying result of ReScript not really having good partial object support.